src/wasm/key_utils.rs

```diff
@@ -87,15 +87,62 @@ pub fn sign(private_key: &str, msg: &str) -> JsValue {
             return JsValue::from_serde(&result).unwrap();
         },
     };
-    sign_with_key(&k, msg, &mut result);
+    sign_message_with_key(&k, msg, None, &mut result);
     JsValue::from_serde(&result).unwrap()
 }
 
-#[allow(non_snake_case)]
-pub(crate) fn sign_with_key(k: &RistrettoSecretKey, msg: &str, result: &mut SignResult) {
-    let (r, R) = RistrettoPublicKey::random_keypair(&mut OsRng);
+/// Generate a Schnorr signature of a challenge that has already been hashed using the given private
+/// key and a specified private nonce. DO NOT reuse nonces. This method is provide for cases where a
+/// public nonce has been used
+/// in the message.
+#[wasm_bindgen]
```

Comment:
I think you meant pre-hashed message, but can you explain what "hashed using the given private key" means? You usually hash with a hashing algorithm.

Reply:
Yeah needs a comma or something. PHRASING.

```
Generate a Schnorr signature of a challenge that has already been hashed, using the given private
```